app/webpacker/components/Results/Records/index.jsx

```diff
@@ -0,0 +1,129 @@
+import { getRecords } from '../api/records';
+import Loading from '../../Requests/Loading';
+import { recordsUrl } from '../../../lib/requests/routes.js.erb';
+import ResultsFilter from '../resultsFilter';
+import SlimRecordTable from './SlimRecordsTable';
+import SeparateRecordsTable from './SeparateRecordsTable';
+
+const ActionTypes = {
+  SET_EVENT: 'SET_EVENT',
+  SET_REGION: 'SET_REGION',
+  SET_RANKING_TYPE: 'SET_RANKING_TYPE',
+  SET_GENDER: 'SET_GENDER',
+  SET_SHOW: 'SET_SHOW',
+};
+
+function parseInitialStateFromUrl(url) {
+  const urlObj = new URL(url);
+  const params = urlObj.searchParams;
+  const region = params.get('region') || 'world';
+  const gender = params.get('gender') || 'All';
+  const show = params.get('show') || '100 persons';
+  const event = params.get('event_id');
+
+  return {
+    event,
+    region,
+    gender,
+    show,
+  };
+}
+
+function filterReducer(state, action) {
+  switch (action.type) {
+    case ActionTypes.SET_EVENT:
+      return { ...state, event: action.payload };
+    case ActionTypes.SET_REGION:
+      return { ...state, region: action.payload };
+    case ActionTypes.SET_GENDER:
+      return { ...state, gender: action.payload };
+    case ActionTypes.SET_SHOW:
+      return { ...state, show: action.payload };
+    default:
+      throw new Error(`Unhandled action type: ${action.type}`);
+  }
+}
+
+export default function Wrapper() {
+  return (
+    <WCAQueryClientProvider>
+      <Rankings />
+    </WCAQueryClientProvider>
+  );
+}
+
+export function Rankings() {
+  const [filterState, dispatch] = useReducer(filterReducer, window.location.href, parseInitialStateFromUrl);
+
+  const filterActions = useMemo(
+    () => ({
+      setEvent: (event) => dispatch({ type: ActionTypes.SET_EVENT, payload: event }),
+      setRegion: (region) => dispatch({ type: ActionTypes.SET_REGION, payload: region }),
+      setGender: (gender) => dispatch({ type: ActionTypes.SET_GENDER, payload: gender }),
+      setShow: (show) => dispatch({ type: ActionTypes.SET_SHOW, payload: show }),
+    }),
+    [dispatch],
+  );
+
+  const {
+    event, region, gender, show,
+  } = filterState;
+
+  const { data, isFetching } = useQuery({
+    queryKey: ['records', event, region, gender, show],
+    queryFn: () => getRecords(event, region, gender, show),
+  });
+
+  useEffect(() => {
+    window.history.replaceState(null, '', recordsUrl(event, region, gender, show));
+  }, [event, region, gender, show]);
+
+  if (isFetching) {
+    return <Loading />;
+  }
+
+  return (
+    <Container fluid>
+      <ResultsFilter
+        filterState={filterState}
+        filterActions={filterActions}
+        isRecords
+        showCategories={['mixed', 'slim', 'separate', 'history', 'mixed history']}
+      />
+      <TableWrapper competitionsById={data.competitionsById} rows={data.rows} show={show} />
+    </Container>
+  );
+}
+
+function TableWrapper({ competitionsById, rows, show }) {
```

Comment:
I don't like how half of the `show`-based logic is handled here, and half is handled in the children i.e. in `RecordsTable`. Handling it all in one spot/level is much nicer. And the higher that level is (and the 'dumber' the children are) the better, generally.

Here, it would be best to either have 5 cases/returns here (1 per show, and if some of those components are very similar then they can just reuse common children), or replace the `show` prop in `RecordsTable` with `resultsAreGroupedByEventId={show !== 'mixed history'}` and other props, so that `RecordsTable` isn't directly aware of `show`. The latter looks complicated/messy in this case, so I'd go with the former. (I'd be happy to directly contribute commits to the PR if that would be helpful.)

Reply:
Feel free to commit to it!

Reply:
I raised a PR, since I couldn't figure out how to push directly.